### ISSUE TITLE
Can't statically detect component wrapped with higher order function

### ISSUE THREAD

**Issue by constgen** [vuejs/eslint-plugin-vue#1319]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: 7.9.0** 
- **eslint-plugin-vue version:7.0.0-beta3** 
- **Node version:12** 
- **Operating System:Windows 10** 

**Please show your full configuration:**

This is only the part related to the issue. The full config is extremely huge.

```json5
rules: {
  "vue/no-unregistered-components": "warn"
}
```

**What did you do?**	

```vue
<template>
  <ModalWindow>
</template>

<script>
import ModalWindow from '../components/ModalWindow'
import { observer } from 'mobx-vue'

export default observer({
  components: {
    ModalWindow
  }
})
</script>
```

**What did you expect to happen?**

ESLint plugin should detect decorated component

**What actually happened?**
`The "ModalWindow" component has been used but not registered`

**Repository to reproduce this issue**
No

I propose 2 options:
- automatically allow higher order functions for Vue Components
- make an exception for Mobx by hardcoding the name `observer`

**Comment by ota-meshi** [comment]
Thank you for this issue.
You can use the special comment `// @vue/component` to detect Vue component.

https://eslint.vuejs.org/user-guide/#how-does-eslint-detect-components

```js
// @vue/component
export default observer({
  components: {
    ModalWindow
  }
})
```

**Comment by constgen** [comment]
I think I have to use the comment too often. It is not an exceptional situation. Almost all components in the application are "observers". Why you are against the proposed options?
